src/main/scala/com/datastax/gatling/plugin/model/DseCqlStatements.scala

```diff
@@ -8,29 +8,35 @@ package com.datastax.gatling.plugin.model
 
 import java.nio.ByteBuffer
 
-import com.datastax.driver.core._
+import com.datastax.oss.driver.api.core.cql.{
+  BatchStatement => BatchS,
+  BatchStatementBuilder => BatchB,
+  BoundStatement => BoundS,
+  BoundStatementBuilder => BoundB,
+  SimpleStatement => SimpleS,
+  SimpleStatementBuilder => SimpleB,
+  _
+}
 import com.datastax.gatling.plugin.exceptions.DseCqlStatementException
 import com.datastax.gatling.plugin.utils.CqlPreparedStatementUtil
 import io.gatling.commons.validation._
-import io.gatling.core.session.{Session, _}
+import io.gatling.core.session._
 
 import scala.collection.JavaConverters._
-import scala.collection.mutable.ArrayBuffer
 import scala.util.{Try, Failure => TryFailure, Success => TrySuccess}
 
-
-trait DseCqlStatement extends DseStatement[Statement] {
-  def buildFromSession(session: Session): Validation[Statement]
-}
+trait DseCqlStatement[T <: Statement[T], B <: StatementBuilder[B,T]] extends DseStatement[B]
```

Comment:
Note that we change the return value of DseStatement.buildFromSession() here.  In prior versions this would return a Statement subclass but as of now it returns a StatementBuilder subclass.  This change facilitates support for immutable statement objects in 2.x; the request Action impl (in this case CqlRequestAction) can operate on a single builder rather than producing lots of intermediate statement objects, one for each defined attribute in DseCqlAttributes.